### PR TITLE
Favorites screen, pepatadicozze.jpg

### PR DESCRIPTION
**Pull request by fourlastor**
Base impl for favorites screen. Right now I'm using schedule's screen views, this is mainly the service layer.

device-2017-03-17-124714

What's missing:

* Date headers
* Views if we decide we want to have different ones
* Theme (right now it's pink)

### CODE REVIEW

**Comment by rock3r** [comment]
it'd be nice to have a screenshot :P

File: app/src/main/java/net/squanchy/favorites/FavoritesPageView.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                        .subscribe(favorites -> updateWith(favorites, this::onEventClicked)));
+
+        subscriptions.add(
+                proximityService.observeProximityEvents()
+                        .observeOn(AndroidSchedulers.mainThread())
+                        .subscribe(this::handleProximityEvent));
+    }
+
+    private void onEventClicked(Event event) {
+        analytics.trackItemSelected(ContentType.SCHEDULE_ITEM, event.id());
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/favorites/FavoritesPageView.java]: Next PR: create a `FAVORITE_ITEM` type plz

File: app/src/main/java/net/squanchy/favorites/FavoritesPageView.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        navigate.toEventDetails(event.id());
+    }
+
+    private void handleProximityEvent(ProximityEvent proximityEvent) {
+        // TODO do something with the event, like showing feedback or opening an event detail
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/favorites/FavoritesPageView.java]: I think we should move this to the activity level instead

File: app/src/main/java/net/squanchy/favorites/FavoritesPageView.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    @Override
+    public void onAttachedToWindow() {
+        super.onAttachedToWindow();
+        subscriptions = new CompositeDisposable();
+        subscriptions.add(
+                service.favorites()
+                        .observeOn(AndroidSchedulers.mainThread())
+                        .subscribe(favorites -> updateWith(favorites, this::onEventClicked)));
+
+        subscriptions.add(
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/favorites/FavoritesPageView.java]: This should be wrapped by the feature flag check

File: app/src/main/java/net/squanchy/favorites/domain/view/Favorites.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+import com.google.auto.value.AutoValue;
+
+import java.util.List;
+
+import net.squanchy.schedule.domain.view.Event;
+
+@AutoValue
+public abstract class Favorites {
+    public static Favorites create(List<Event> events) {
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/favorites/domain/view/Favorites.java]: cmd-alt-l plz

File: app/src/main/java/net/squanchy/schedule/view/EventsAdapter.java [hunk 5]
```diff
@@ -24,15 +24,12 @@
         int TYPE_OTHER = 1;
     }
 
-    private final Context context;
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/schedule/view/EventsAdapter.java:27]: 💯

File: app/src/main/java/net/squanchy/favorites/FavoritesPageView.java (hunk 2 above)
Comment (fourlastor) [reply app/src/main/java/net/squanchy/favorites/FavoritesPageView.java]: Yep agreed

File: app/src/main/java/net/squanchy/favorites/domain/view/Favorites.java (hunk 4 above)
Comment (fourlastor) [reply app/src/main/java/net/squanchy/favorites/domain/view/Favorites.java]: e52f80d

**Review by rock3r — APPROVED** [review]

File: app/src/main/java/net/squanchy/favorites/FavoritesPageView.java (hunk 2 above)
Comment (fourlastor) [reply app/src/main/java/net/squanchy/favorites/FavoritesPageView.java]: 252059f

File: app/src/main/java/net/squanchy/favorites/FavoritesPageView.java (hunk 3 above)
Comment (fourlastor) [reply app/src/main/java/net/squanchy/favorites/FavoritesPageView.java]: 252059f

**Review by rock3r — APPROVED** [review]
LGTM 👍

**Pull request merged**